### ISSUE TITLE
add random number seed parameter

### ISSUE THREAD

**Issue by rpreen** [rpreen/xcsf#85]
Currently, the random number seed is set via a Python function `def seed(value: int) -> None`. Changing this to a parameter `random_state` will allow it to be set through JSON and therefore the stand-alone executable; it will also enable being set via a future `set_params()` and constructor args, making it more similar to sklearn.
